tests/test_events/handler/addTab.test.js

```diff
@@ -0,0 +1,138 @@
+        console.log(`Creating directories: repoPath=${repoPath}, tabsRepo=${tabsRepo}`);     
+        jest.clearAllMocks();
+        untildify.mockReturnValue(baseRepoPath);
+        const returnedPath = untildify();
+        console.log(`untildify returned: ${returnedPath}`);
+    });
+
+    afterEach(() => {
+        fs.rmSync(repoPath, { recursive: true, force: true });
+        fs.rmSync(baseRepoPath, { recursive: true, force: true });
+    });
+    
+    // Skips case when repo exists
+    test("Should be skipped if repo does not exist",() => {
+        const handler = new tabEventHandler();
+        handler.handleTabChangeEvent();
+        // Verify no tab file should be generated
+        assertNoTabsRecorded(tabsRepo)   
+        });
+
+    //  Skips cases where repo exists, but is not connected
+    test("Should skip if repo exists, but is not connected", () => {
+        const handler = new tabEventHandler(repoPath);
+        handler.handleTabChangeEvent();
+        // Verify no tab file should be generated
+        assertNoTabsRecorded(tabsRepo)    
+    })
+
+    // Skip case where repo is connected, but user account is not valid
+    test("Should skip if repo is connected, but user account is not valid", () => {
+        configUtil.addRepo();
+        addUser(baseRepoPath, false);
+        const handler = new tabEventHandler(repoPath);
+        handler.handleTabChangeEvent(false);
+        // Verify no tab file should be generated
+        assertNoTabsRecorded(tabsRepo)       
+    })
+
+    // Skip case where user account is not valid, but repo is not connected 
+    test("Should skip if user account is valid, but repo is not connected", () => {
+        addUser(baseRepoPath, true);
+        const handler = new tabEventHandler(repoPath);
+        handler.handleTabChangeEvent();
+        // Verify no tab file should be generated
+        assertNoTabsRecorded(tabsRepo)      
+    })
+
+    // Should skip if not opened/closed event
+    test("Should skip if not opened/closed event", () => {
+        // Repo is connected & user account is also valid
+        configUtil.addRepo();
+        addUser(baseRepoPath, true);
+        const handler = new tabEventHandler(repoPath);
+        // Pass isTabEvent=false, which means changeEvent.changed.length > 0
+        handler.handleTabChangeEvent(false);
+        // Verify no tab file should be generated
+        assertNoTabsRecorded(tabsRepo)    
+    })
+
+    // Skip if invalid repo id
+    test("Should skip if repoId doesn't exist", () => {
+        const invalid_repo_path = (repoPath).slice(0,5);
+        const configUtil_2 = new Config(invalid_repo_path, configPath);
+        addUser(baseRepoPath, true);
+        const handler = new tabEventHandler(invalid_repo_path);
+        handler.handleTabChangeEvent(true);
+        // Verify no tab file should be generated
+        assertNoTabsRecorded(tabsRepo)    
+    })
+
+    test("Tabs data in positive case", async () => {
```

Comment:
@saadberry Positive case should assert that a file is added to the buffer correctly, then read the file from buffer and then assert its structure as also correct